bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -351,12 +363,29 @@ public String toString() {
         return toHexString();
     }
 
-    // see https://docs.oracle.com/javase/6/docs/platform/serialization/spec/output.html
+    /**
+     * Write the replacement object.
+     *
+     * <p>
+     * See https://docs.oracle.com/javase/6/docs/platform/serialization/spec/output.html
+     * </p>
+     *
+     * @return a proxy for the document
+     */
     private Object writeReplace() {
         return new SerializationProxy(this);
     }
 
-    // see https://docs.oracle.com/javase/6/docs/platform/serialization/spec/input.html
+    /**
+     * Prevent normal serialization.
+     *
+     * <p>
+     * See https://docs.oracle.com/javase/6/docs/platform/serialization/spec/input.html
```

Comment:
[optional, ignore if consider to be too much of a hassle]

Let's update to https://docs.oracle.com/en/java/javase/17/docs/specs/serialization/input.html#the-readobject-method. This comment is applicable to all other similar changes.

Reply:
As above